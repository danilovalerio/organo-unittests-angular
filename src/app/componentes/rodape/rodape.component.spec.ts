import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RodapeComponent } from './rodape.component';

describe('RodapeComponent', () => {
  let component: RodapeComponent;
  let fixture: ComponentFixture<RodapeComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RodapeComponent],
    });

    fixture = TestBed.createComponent(RodapeComponent);
    component = fixture.componentInstance;
  });

  it('deveria ser criado', () => {
    expect(component).toBeTruthy();
  });

  it('deveria definir as propriedades alt e src', () => {
    expect(component.alt).toBeDefined();
    expect(component.src).toBeDefined();
  });

  it('deveria renderizar o conteudo baseado nas propriedades src e alt', () => {
    component.src = 'http://example.com/test-imagem.jpg';
    component.alt = 'Imagem de teste';

    expect(component).toMatchSnapshot();
  });
});
