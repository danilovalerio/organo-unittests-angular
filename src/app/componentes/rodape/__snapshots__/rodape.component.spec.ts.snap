// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RodapeComponent deveria renderizar o conteudo baseado nas propriedades src e alt 1`] = `
RodapeComponent {
  "__ngContext__": 4,
  "alt": "Imagem de teste",
  "src": "http://example.com/test-imagem.jpg",
}
`;
